src/gwc/src/main/java/org/geoserver/gwc/layer/DefaultTileLayerCatalog.java

```diff
@@ -292,7 +293,7 @@ private GeoServerTileLayerInfoImpl depersist(final Resource res) throws IOExcept
             LOGGER.fine("Depersisting GeoServerTileLayerInfo from " + res.path());
         }
         GeoServerTileLayerInfoImpl info;
-        try(Reader reader = new InputStreamReader(res.in(), "UTF-8")) {
+        try(Reader reader = new InputStreamReader(new ByteArrayInputStream(res.getContents()), "UTF-8")) {
```

Comment:
It's a good performance improvement. The getContents() call I mean, which returns a byte[].
Do you want to have a utility method to wrap the byte[] in a ByteArrayInputStream?